xray/audit/java/gradle.go

```diff
@@ -168,6 +170,46 @@ func (dtp *depTreeManager) createDepTreeScript() (tmpDir string, err error) {
 	return tmpDir, errorutils.CheckError(os.WriteFile(filepath.Join(tmpDir, depTreeInitFile), []byte(depTreeInitScript), 0666))
 }
 
+// getRemoteRepos constructs the sections of Artifactory's remote repositories in the gradle-dep-tree init script.
+// releasesRepoName - name of the remote repository that proxies https://releases.jfrog.io
+// depsRemoteRepo - name of the remote repository that proxies the dependencies server, e.g. maven central.
+// server - the Artifactory server details on which the repositories reside in.
+// Returns the constructed sections.
+func getRemoteRepos(releasesRepo, depsRepo string, server *config.ServerDetails) (string, string, error) {
+	constructedReleasesRepo, err := constructReleasesRemoteRepo(releasesRepo, server)
+	if err != nil && err != errEmptyReleasesRepo {
+		return "", "", err
+	}
+
+	constructedDepsRepo, err := constructDepsRemoteRepo(depsRepo, server)
+	if err != nil && err != errEmptyDepsRepo {
+		return "", "", err
+	}
+	return constructedReleasesRepo, constructedDepsRepo, nil
+}
+
+func constructReleasesRemoteRepo(releasesRepo string, server *config.ServerDetails) (string, error) {
+	if releasesRepo == "" {
+		// Try to get releases repository from the environment variable
+		releasesRepo = os.Getenv(coreutils.ReleasesRemoteEnv)
+		_, repoName, err := coreutils.SplitRepoAndServerId(releasesRepo, coreutils.ReleasesRemoteEnv)
+		if err != nil {
+			return "", errEmptyReleasesRepo
```

Comment:
It is best not to loose the value of `err`. You can use -
```
return "", errors.Join(errEmptyReleasesRepo, err)
```
`errors.Join` was added in go 1.20.